File: lib/enums.dart
import 'package:flutter/cupertino.dart';

enum ClipboardContentType {
  text,
  image,
  file,
  unknown;

  static ClipboardContentType parse(String value) =>
      ClipboardContentType.values.firstWhere(
        (e) => e.name.toUpperCase() == value.toUpperCase(),
        orElse: () {
          debugPrint("ClipboardContentType '$value' unknown");
          return ClipboardContentType.unknown;
        },
      );
}

enum EnvironmentType {
  shizuku,
  root,
  androidPre10,
  none;

  static EnvironmentType parse(String value) =>
      EnvironmentType.values.firstWhere(
        (e) => e.name == value,
        orElse: () {
          debugPrint("ClipboardContentType '$value' unknown");
          return EnvironmentType.none;
        },
      );
}

File: android/src/main/kotlin/top/coclyun/clipshare/clipboard_listener/enums.kt
package top.coclyun.clipshare.clipboard_listener

enum class EnvironmentType {
    shizuku,
    root,
    androidPre10
}

enum class EventEnum {
    onChanged, comment, fatal, eof
}

enum class ClipboardContentType {
    Text, Image;

    companion object {
        fun parse(name: String): ClipboardContentType {
            for (v in values()) {
                if (v.name.lowercase() == name.lowercase()) {
                    return v
                }
            }
            throw IllegalArgumentException("no such element $name")
        }
    }
}

enum class ClipboardListeningWay {
    logs, hiddenApi;
}